keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -535,29 +547,79 @@ Disk-pressure eviction:
   - `reason=Evicted`
   - `message=The node was low on resource: ephemeral-storage. (...)`
 - Container status:
-  - `state=Ternminated`
+  - `state=Terminated`
   - `exitCode=137`
   - `reason=Error`
-- Retriable: Unclear, excessive disk usage suggests a bug or misconfiguration.
-  However, a restart on another node may succeed
+- Retriable: Yes
+
+##### Container memory limit exceeded
 
-##### OOM kill
+Linux:
 
 - Reproduction: We run a job with a pod which attempts to allocate more
   memory than constrained in the container spec by `resources.limits.memory`
-- Comments: handled by kubelet
+- Comments: handled by kubelet in `kubelet/kubelet_pods.go` which merges-in the status update from container runtime
 - Pod status:
   - status: OOMKilled
   - `phase=Failed`
   - `reason=`
   - `message=`
 - Container status:
-  - `state=Ternminated`
+  - `state=Terminated`
   - `exitCode=137`
   - `reason=OOMKilled`
-- Retriable: Unclear, but if occurs when
-  `resources.requests.memory=resources.limits.memory` it strongly suggests
-  a software bug.
+- Retriable: No
+
+Windows:
+
+- Reproduction: We run a job with a pod which attempts to allocate more
+  memory than constrained in the container spec by `resources.limits.memory`
+- Comments: handled by kubelet in `kubelet/kubelet_pods.go` which merges-in the status update from container runtime
+- Pod status:
+  - status: Error
+  - `phase=Failed`
+  - `reason=`
+  - `message=`
+- Container status:
+  - `state=Terminated`
+  - `exitCode=1`
+  - `reason=Error`
+- Retriable: No
+
+##### Graceful node shutdown
+
+- Reproduction: The node needs to be started with positive `shutdownGracePeriod`,
+  for example `shutdownGracePeriod=30s`. We run a job with a long-running pod,
+  then we prepare the node for shutdown by the command:
+  `dbus-send --system --type=signal /org/freedesktop/login1 org.freedesktop.login1.Manager.PrepareForShutdown boolean:"true"`,
+  which similulates graceful nodes shutdown for maintenance purposes.
+- Comments: handled by kubelet in `kubelet/nodeshutdown/nodeshutdown_manager_linux.go`
```

Comment:
is there a windows version? do we need to add this case to non-goals too?

Reply:
There is prepared file for "shutdown_manager_others.go" (currently has no logic: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/nodeshutdown/nodeshutdown_manager_others.go),
so I assume the events are not handled on Windows. Added a non-goal point.